design/FY2021/NFP-Waterside-Economizer.md

```diff
@@ -0,0 +1,795 @@
+        \note For all dehumidification controls, the max
+        \note humidity setpoint on the Sensor Node is used.
+        \note SetpointManager:SingleZone:Humidity:Maximum,
+        \note SetpointManager:MultiZone:Humidity:Maximum, or
+        \note SetpointManager:MultiZone:MaximumHumidity:Average, and
+        \note SetpointManager:OutdoorAirPretreat (optional) objects.
+    A8, \field Run on Sensible Load
+        \type choice
+        \key Yes
+        \key No
+        \default Yes
+        \note If Yes, unit will run if there is a sensible load.
+        \note If No, unit will not run if there is only a sensible load.
+        \note Dehumidification controls will be active if specified.
+    A9; \field Run on Latent Load
+        \type choice
+        \key Yes
+        \key No
+        \default No
+        \note If Yes, unit will run if there is a latent load.
+        \note even if there is no sensible load.
+        \note If No, unit will not run only if there is a latent load.
+        \note Dehumidification controls will be active if specified.
+
+Several considerations need to be resolved for this new object.
+
+ - Should a Sensor Node Name input field be included, or force use of the object or coil outlet node (recommended and agreed to via comments)
+ - Should a controller tolerance input field be included or assume a typical value of 0.001 for temperature control and 0.00001 for humidity control (no comments, assumed OK)
+ - If the CoilSystem objects are consolidated, additional inputs would be needed for Use Outdoor Air DX Cooling Coil (A11) and Outdoor Air DX Cooling Coil Leaving Minimum Air Temperature (N1) to replicate all inputs for CoilSystem:Cooling:DX. Additional inputs would also be required for heat exchanger type and name. (discouraged via comments)
+
+## Testing/Validation/Data Sources ##
+
+Compare results of existing coil types with this new object. Also test performance of the new waterside economizer model with object attached to a condenser loop and upstream of a UnitarySystem.
+
+## Input Output Reference Documentation ##
+
+Add new IO section for new object.
+
+## Outputs Description ##
+
+	Cooling Coil Water Flow Rate
+	Cooling Coil Water Flow Fraction
+	Cooling Coil Total Cooling Rate
+	Cooling Coil Total Cooling Energy
+	Cooling Coil Source Side Heat Transfer Energy
+	Cooling Coil Sensible Cooling Rate
+	Cooling Coil Sensible Cooling Energy
+	Cooling Coil Latent Cooling Rate
+	Cooling Coil Latent Cooling Energy
+	Cooling Coil Water-Side Economizer Status
+
+## Engineering Reference ##
+
+Add new Engineering section for new object.
+
+## Example File and Transition Changes ##
+
+No transition required. New example file(s) will be included.
+
+## References ##
+
+None.
+
+## Design Document ##
+
+The UnitarySystem is the only air-side parent that uses the pointer/factory method. The pointer of the class HVACSystemData. The same appraoach will be used for CoilWaterSystems.
```

Comment:
@rraustad has demonstrated a method for the AirLoopHVACX:UnitarySystem code to simulate CoilSystem:Cooling:DX in #8729. It should be possible to do the same here with the small addition of support for the cutoff temperature difference. That means this feature will not need a new source file, just a new input function in UnitarySystem. This may reduce (or eliminate) the need for changes in other source files as well.

Reply:
@mjwitte It is not clear to me when you say "this feature will not need a new source file".  I think we need separate files (CoilWaterSystems.hh and CoilWaterSystems.cc) to hold the data structure definitions and all the functions we need such as getInput(), Initialization(), Control(), simulate(), etc for the CoilWaterSystem. Currently "CoilSystem:Cooling:DX" resides, in HVACDXSystem files.

Reply:
Except for the fact that #8729 has not yet been merged or perfected to the point where it can be merged (i.e., that branch will definitely have changes). Not sure how this impacts the need for new source code.

Reply:
My vision of this is that CoilSystem:Cooling:Water would be just an abbreviated synonym for AirloopHVAC:UnitarySystem. It would instantiate a unitarysystem internally and that code will do all the work. With that approach, there's no need for CoilWaterSystem.hh or cc. I realize the concept isn't quite baked yet for CoilSystem:Cooling:DX, but I can smell the cake in the oven.

Reply:
I might suggest a path forward for this feature would be to temporarily add the temperature cutoff field to the end of AirloopHVAC:UnitarySystem and set up an example file based on unitary system, add support for the new control to unitarysystem, and then when the cake is baked (mmm, citrus, lemon?) a getinput for CoilSystem:Cooling:Water can be added to unitarysystem and then it's done (mostly).

Reply:
If this is the preferred option, I am OK with it. Is every body on board with this approach? @rraustad @EnergyArchmage @Myoldmopar

Reply:
@Nigusse I think "I might suggest a path forward" lacks sufficient background on using UnitarySystem to model other system types (e.g., CoilSystem:Cooling:DX, and in your case CoilSystem:Cooling:Water). The new object CoilSystem:Cooling:Water will need to be read in by someone (i.e., some model in E+). Having the UnitarySystem model cooling coil type = CoilSystem:Cooling:Water would typically mean calling a Sim function and other mining functions in another cc module (e.g., getCoilName). @mjwitte is suggesting you NOT create a new module. Which means you will need to read that object in using UnitarySystem. Not sure @mjwitte thought this suggestion all the way through. Either a new module is needed or a new getWaterCoilSystemData is needed (someone has to read the object). Similar to getDXCoilSystemData in #8729. But that method is not yet ready. It's getting pretty close just not fully fleshed out yet.

So to actually do what @mjwitte suggests would mean you need to create a getWaterCoilSystemData function and translate those inputs to UnitarySystem. This is really scary considering #8729 is not yet completed/merged. I am OK doing it this way it will just take some coordination between developers.

Reply:
> I might suggest a path forward for this feature would be to temporarily add the temperature cutoff field to the end of AirloopHVAC:UnitarySystem and set up an example file based on unitary system, add support for the new control to unitarysystem, and then when the cake is baked (mmm, citrus, lemon?) a getinput for CoilSystem:Cooling:Water can be added to unitarysystem and then it's done (mostly).

OK. Implemented @mjwitte suggestion but ended up with two new fields: (1) "Yes" or "No" choice field for triggering UnitarySystem as a water side economizer coil system and (2) a field for specifying the temperature offset value.  Also added the necessary controls, and a new report variable that tell the water side economizer operation status.

Tested the code in a single zone model in Miami and Chicago weather locations and it is working as expected. I will Post  sample results shortly.

Reply:
I was thinking that if temperature offset = 0 then it would act at a regular cooling coil.

Reply:
I am not sure what you mean as a regular cooling coil? This cooling coil is connected to the demand side of the condenser loop and this cooling coil can operate only when the coil entering water temperature is less than the coil entering air temperature. And what the temp offset does is it allows the coil to operate when there is larger driving potential (temp diffs) and there is cooling load. Can you explain what you mean a regular cooling coil?

Reply:
> Well, I forgot it was connected to the condenser loop. But if the coil is connected to the condenser loop then you already know it will be used as an economizer coil, don't you? If the user input water side economizer coil = No what would that mean?

I am just testing how the UnitarySystem connected to condenser loop and placed on airloop operate as a water-side economizer . "No" means that the UniatrySystem is not operated or controlled as a waterside economizer. If I can determine that the UnitarySystem is used as a water side economizer from other existing inputs, then this input field is not required. I have think about it.

Reply:
@Nigusse I haven't looked at the code changes yet, but it seems that using the coilsystem as a water-side economizer shouldn't change the controls at all. If the incoming water temperature exceeds the incoming air temp minus the offset then it tries to operate, if not, then it doesn't. If there is a reason the unitarsystem needs to know this, then I would prefer an explicit input field rather than some automated check (which will likely be mysterious to users and likely not work for some use case that wasn't anticipated).

Reply:
> And that can be a field only for CoilSystem:Water if we want. It doesn't necessarily have to remain in place for AirloopHVAC:UnitarySystem.

@mjwitte I agree these input fields must reside in CoilSystem:Water but for now I need it to explicitly tell unitary system how it is going to operate.

Reply:
Currently I am testing the changes in PR #8766 but will pull in the changes here at some point.